File: mobile/src/main/java/melquelolea/vefexchange/VefExchangeWidget.java
package melquelolea.vefexchange;

import android.app.AlarmManager;
import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.RemoteViews;

import melquelolea.vefexchange.data.UpdateDataService;

/**
 * Implementation of App Widget functionality.
 */
public class VefExchangeWidget extends AppWidgetProvider {

    private static final String TAG = VefExchangeWidget.class.getSimpleName();
    private static final int JOB_ID = 44;

    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        //Start the background service to update the widgets
        context.startService(new Intent(context, UpdateDataService.class));
        JobScheduler jobScheduler =
                (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);

        long interval = AlarmManager.INTERVAL_FIFTEEN_MINUTES;
        JobInfo job = new JobInfo.Builder(JOB_ID,
                new ComponentName(context.getPackageName(),
                        UpdateDataService.class.getName()))
                .setPersisted(true)
                .setPeriodic(interval)
                .build();

        jobScheduler.schedule(job);

        Log.d(TAG, "setting scheduled job for: " + interval);
    }

    @Override
    public void onEnabled(Context context) {
        // Enter relevant functionality for when the first widget is created
    }

    @Override
    public void onDisabled(Context context) {
        // Enter relevant functionality for when the last widget is disabled
    }
}

File: mobile/src/main/java/melquelolea/vefexchange/widget/VefExchangeWidget.kt
package melquelolea.vefexchange.widget

import android.app.AlarmManager
import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.util.Log

import melquelolea.vefexchange.services.UpdateDataService

/**
 * Implementation of App Widget functionality.
 */
class VefExchangeWidget : AppWidgetProvider() {

    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
        //Start the background service to update the widgets
        context.startService(Intent(context, UpdateDataService::class.java))
        val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler

        val interval = AlarmManager.INTERVAL_FIFTEEN_MINUTES
        val job = JobInfo.Builder(JOB_ID,
                ComponentName(context.packageName,
                        UpdateDataService::class.java.name))
                .setPersisted(true)
                .setPeriodic(interval)
                .build()

        jobScheduler.schedule(job)

        Log.d(TAG, "setting scheduled job for: " + interval)
    }

    override fun onEnabled(context: Context) {
        // Enter relevant functionality for when the first widget is created
    }

    override fun onDisabled(context: Context) {
        // Enter relevant functionality for when the last widget is disabled
    }

    companion object {

        private val TAG = VefExchangeWidget::class.java.simpleName
        private val JOB_ID = 44
    }
}

